src/flask/config.py

```diff
@@ -256,12 +257,18 @@ def from_file(
         :param silent: Ignore the file if it doesn't exist.
         :return: ``True`` if the file was loaded successfully.
 
+        .. versionchanged:: 2.3.0
+            Added optional boolean argument "text" (False by default).
+            You can now use .toml config file by specifying "text=True".
+
         .. versionadded:: 2.0
         """
         filename = os.path.join(self.root_path, filename)
 
         try:
             with open(filename) as f:
+                if text:
```

Comment:
This should change what `open` does, not try to access `buffer` after the fact (someone else pointed out this wasn't documented as a public API).

Also, it's backwards, `text=True`, the default, should result in a text file, `False` should result in a binary file.